app/mender.go

```diff
@@ -466,6 +466,34 @@ func (m *Mender) TransitionState(to State, ctx *StateContext) (State, bool) {
 	return transitionState(to, ctx, m)
 }
 
+func spinEventLoop(c *ControlMapPool, ctx *StateContext, to State) (State, bool) {
```

Comment:
So far untested, but I'd want some feedback on the approach, is this sort of correct?
i.e., passing in the `Transition` name from all transitions to `QueryAndUpdate`?
and returning the `NewUpdateReportState` in case of error?

Reply:
The principle is very much in line with how I envisioned it, though there are some details that need to be worked out. See individual comments.